crates/router/src/core/payment_methods/vault.rs

```diff
@@ -551,3 +568,140 @@ pub async fn delete_tokenized_data(
+        .attach_printable("Failed while getting delete tokenize data schedule time")?;
+
+    let process_tracker_entry = storage::ProcessTrackerNew {
+        id: format!("{}_{}_{}", runner, task, lookup_key),
+        name: Some(String::from(task)),
+        tag: vec![String::from("BASILISK-V3")],
+        runner: Some(String::from(runner)),
+        retry_count: 0,
+        schedule_time,
+        rule: String::new(),
+        tracking_data,
+        business_status: String::from("Pending"),
+        status: enums::ProcessTrackerStatus::New,
+        event: vec![],
+        created_at: current_time,
+        updated_at: current_time,
+    };
+    let response = db
+        .insert_process(process_tracker_entry)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable_lazy(|| {
+            format!(
+                "Failed while inserting task in process_tracker: lookup_key: {}",
+                lookup_key
+            )
+        })?;
+    Ok(response)
+}
+
+#[cfg(feature = "basilisk")]
+pub async fn start_tokenize_data_workflow(
+    state: &routes::AppState,
+    tokenize_tracker: &storage::ProcessTracker,
+) -> Result<(), errors::ProcessTrackerError> {
+    let db = &*state.store;
+    let delete_tokenize_data = serde_json::from_value::<storage::TokenizeCoreWorkflow>(
+        tokenize_tracker.tracking_data.clone(),
+    )
+    .into_report()
+    .change_context(errors::ApiErrorResponse::InternalServerError)
+    .attach_printable_lazy(|| {
+        format!(
+            "unable to convert into DeleteTokenizeByTokenRequest {:?}",
+            tokenize_tracker.tracking_data
+        )
+    })?;
+
+    let delete_resp = delete_tokenized_data(state, &delete_tokenize_data.lookup_key).await;
+    match delete_resp {
+        Ok(resp) => {
+            if resp == "Ok" {
+                logger::info!("Card From locker deleted Successfully");
+                //mark task as finished
+                let id = tokenize_tracker.id.clone();
+                tokenize_tracker
+                    .clone()
+                    .finish_with_status(db, format!("COMPLETED_BY_PT_{id}"))
+                    .await?;
+            } else {
+                logger::error!("Error: Deleting Card From Locker : {}", resp);
+                retry_delete_tokenize(db, &delete_tokenize_data.pm, tokenize_tracker.to_owned())
+                    .await?;
+            }
+        }
+        Err(err) => {
+            logger::error!("Err: Deleting Card From Locker : {}", err);
+            retry_delete_tokenize(db, &delete_tokenize_data.pm, tokenize_tracker.to_owned())
+                .await?;
+        }
+    }
+    Ok(())
+}
+
+#[cfg(feature = "basilisk")]
+pub async fn get_delete_tokenize_schedule_time(
+    db: &dyn db::StorageInterface,
+    pm: &enums::PaymentMethod,
+    retry_count: i32,
+) -> Result<Option<time::PrimitiveDateTime>, errors::ProcessTrackerError> {
```

Comment:
this function can be just `time::PrimitiveDateTime`

Reply:
should be `Option<time::PrimitiveDateTime>` in case `RETRIES_EXCEEDED`